pyproject.toml

```diff
@@ -1,3 +1,70 @@
 [build-system]
-requires = ["setuptools"]
+requires = ["setuptools >= 61.0"]
 build-backend = "setuptools.build_meta"
+
+[project]
+name = "xmlschema"
+version = "3.4.3"
+description = "An XML Schema validator and decoder"
+readme = "README.rst"
+license = {text = "MIT"}
+requires-python = ">=3.8"
+authors = [
+    { name = "Davide Brunato", email = "brunato@sissa.it" },
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Information Technology",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Programming Language :: Python :: 3.14",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Programming Language :: Python :: Implementation :: PyPy",
+    "Topic :: Software Development :: Libraries",
+    "Topic :: Text Processing :: Markup :: XML",
+]
+dependencies = [
+    "elementpath>=4.4.0, <5.0.0",
+]
+
+[project.optional-dependencies]
```

Comment:
`elementpath` is a dependency of the library, so I did not include it in every optional dependency group. Also I used a recursive call here.

The `dev` includes `codegen` and `docs`. Actually I was unsure about the `codegen` group. What do you think?

Reply:
`codegen` requires only elementpath and jinja2, it's correct